core/src/main/java/org/apache/iceberg/io/ImmutableStorageCredential.java

```diff
@@ -0,0 +1,359 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.io;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import com.google.errorprone.annotations.Var;
+import java.io.ObjectStreamException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.CheckReturnValue;
+import javax.annotation.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+import javax.annotation.concurrent.Immutable;
+import javax.annotation.concurrent.NotThreadSafe;
+import org.apache.iceberg.util.SerializableMap;
+import org.immutables.value.Generated;
+
+/**
+ * This was copied from the generated Immutable implementation of {@link StorageCredential} with the
+ * one difference that the underlying Map is not unmodifiable but rather a {@link SerializableMap}
+ * so that Ser/De properly works with Kryo.
+ *
+ * <p>Use the builder to create immutable instances: {@code ImmutableStorageCredential.builder()}.
+ */
+@Generated(from = "StorageCredential", generator = "Immutables")
```

Comment:
Is there anything Immutables does with this annotation?  It seems like something they would include but maybe we shouldn't?

Reply:
I think we can actually remove both annotations